plugins/common/socket/socket.go

```diff
@@ -96,74 +99,62 @@ func (cfg *Config) NewSocket(address string, splitcfg *SplitConfig, logger teleg
 }
 
 func (s *Socket) Setup() error {
+	if s.MaxParallelParsers == 0 {
+		cpus, err := cpu.Counts(true)
+		if err != nil {
+			return err
+		}
+
+		s.MaxParallelParsers = max(cpus/2, 1)
+	}
```

Comment:
I think zero is a perfectly valid indication of "no parallelization"! Please don't use magic here!